### PR TITLE
Added product-listing page

### PR DESCRIPTION
**Pull request by sanket4120**
Added product listing page & filters
files to review:
- components: filters
- pages: shop
- reducers: productReducers
- utils
- context: productsContext

### CODE REVIEW

File: src/actions/categoryActions.js [hunk 1]
```diff
@@ -1,28 +1,16 @@
...
-
-    getCategories();
-  }, [dispatch]);
-
-  return state;
+export const getCategories = async (dispatch) => {
+  dispatch({ type: CATEGORY_REQUEST });
+  try {
+    const res = await axios.get('/api/categories');
+    dispatch({ type: CATEGORY_SUCCESS, payload: res.data.categories });
```
Comment (govindappaarun) [inline src/actions/categoryActions.js:12]: Suggestion: if you add a prefix FETCH it will be more readable, eg: FETCH_CATEGORY_SUCCESS

**Review by govindappaarun — APPROVED** [review]
Looks good, 
Good naming conventions and code structure

File: src/server.js [hunk 2]
```diff
@@ -64,6 +66,8 @@ export function makeServer({ environment = 'development' } = {}) {
       // products routes (public)
```
Comment (GeetaPogul) [inline src/server.js:66]: remove if you don't need commented code

**Review by GeetaPogul — APPROVED** [review]
code looks good and clean.

**Pull request merged**